aiida_quantumespresso/cli/workflows/matdyn/base.py

```diff
@@ -14,7 +12,11 @@
 
 @cmd_launch.command('matdyn-base')
 @options.CODE(required=True, type=types.CodeParamType(entry_point='quantumespresso.matdyn'))
-@options.CALCULATION(required=True)
+@options.DATUM(
+    required=True,
+    type=types.DataParamType(sub_classes=('aiida.data:quantumespresso.forceconstants',)),
+    help='A ForceconstantsData node produced by a `Q2rCalculation`'
```

Comment:
* After PR #386, these should become `'aiida.data:quantumespresso.force_constants'` and `ForceConstantsData` respectively.
* Using a `datum` instead of a `calculation` makes using this launcher a bit less easy, which is unfortunate, but it's certainly cleaner code.

Reply:
I also initially thought that it was less easy to use, but after some thinking I am not sure. Since anyway before being able to launch it you need to run a `Q2rCalculation`, if you do it through the CLI as well, at the end the `ForceConstantsData` output node will be printed with a pk. If you want to restart later from a `Q2rCalculation` pk, it only takes `verdi process show` to get the correct output node. So I think this is fine in the end